corebuild/integration/ILLink.Tasks/LinkTask.cs

```diff
@@ -59,51 +85,116 @@ public class ILLink : Task
 		/// </summary>
 		public bool DumpDependencies { get; set; }
 
-		public override bool Execute()
+		public override bool Execute ()
 		{
-			string[] args = GenerateCommandLineCommands();
-			var argsString = String.Join(" ", args);
-			Log.LogMessageFromText($"illink {argsString}", MessageImportance.Normal);
-			int ret = Mono.Linker.Driver.Main(args);
+			string applicationAssemblyAction;
+			string usedApplicationAssemblyAction = UsedApplicationAssemblyAction.ItemSpec;
+			string unusedApplicationAssemblyAction = UnusedApplicationAssemblyAction.ItemSpec;
+			if (!GetAssemblyAction (usedApplicationAssemblyAction.ToLower (), unusedApplicationAssemblyAction.ToLower (), out applicationAssemblyAction)) {
+				Log.LogError ("Unsupported combination of application assembly actions: {0}, {1}.",
+							  usedApplicationAssemblyAction, unusedApplicationAssemblyAction);
+				return false;
+			}
+
+			string netCoreFrameworkAssemblyAction;
+			string usedNetCoreFrameworkAssemblyAction = UsedNetCoreFrameworkAssemblyAction.ItemSpec;
+			string unusedNetCoreFrameworkAssemblyAction = UnusedNetCoreFrameworkAssemblyAction.ItemSpec;
+			if (!GetAssemblyAction (usedNetCoreFrameworkAssemblyAction.ToLower (), unusedNetCoreFrameworkAssemblyAction.ToLower (), out netCoreFrameworkAssemblyAction)) {
+				Log.LogError ("Unsupported combination of .NET Core Framework assembly actions: {0}, {1}.",
+							  usedNetCoreFrameworkAssemblyAction, unusedNetCoreFrameworkAssemblyAction);
+				return false;
+			}
+
+			string [] args = GenerateCommandLineCommands (applicationAssemblyAction, netCoreFrameworkAssemblyAction);
+			var argsString = String.Join (" ", args);
+			Log.LogMessageFromText ($"illink {argsString}", MessageImportance.Normal);
+			int ret = Mono.Linker.Driver.Main (args);
 			return ret == 0;
 		}
 
-		private string[] GenerateCommandLineCommands()
+		bool GetAssemblyAction (string usedAssemblyAction, string unusedAssemblyAction, out string assemblyAction)
 		{
-			var args = new List<string>();
+			assemblyAction = "illegal";
+			if ((unusedAssemblyAction != usedAssemblyAction) && (unusedAssemblyAction != "delete")) {
+				return false;
+			}
+
+			switch (usedAssemblyAction) {
+			case "trim":
+				assemblyAction = "link";
+				break;
+
+			case "copy":
+				assemblyAction = (unusedAssemblyAction == "Delete") ? "copyused" : "copy";
+				break;
+
+			case "addbypassngen":
+				assemblyAction = (unusedAssemblyAction == "delete") ? "addbypassngenused" : "addbypassngen";
+				break;
+
+			case "skip":
+				if (unusedAssemblyAction != usedAssemblyAction) {
+					return false;
+				}
+				assemblyAction = "skip";
+				break;
+
+			default:
+				return false;
+			}
+
+			return true;
+		}
```

Comment:
Can you explain the rationale for this validation? This is just setting reasonable default modes, right? I think this should live in the msbuild targets to make it easy to override.